src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1179,6 +1187,8 @@ internal TextLoader(IHostEnvironment env, Options options = null, IMultiStreamSo
                 _flags |= OptionFlags.AllowSparse;
             if (options.AllowQuoting && options.ReadMultilines)
                 _flags |= OptionFlags.ReadMultilines;
+            if (options.ImputeEmptyFloats)
```

Comment:
I took a look of your PR and realize that there is some legacy with TextLoader(impute empty with null). So do we want to treat it like when users want imputation, we fill NaNs for empty strings. Other than that, we throw errors when parsing whatever to NaNs

Reply:
Is this suggestion related to the issue you've opened (#5203)?  If so, I think it's better to address that issue in a separate PR, and with a different option, as I think they're different use cases. It should be possible for a user to load empty as NaN, but still be able to load un-parsable strings also as NaN.

---
In reply to: [436910845](https://github.com/dotnet/machinelearning/pull/5198#discussion_r436910845) [](ancestors = 436910845)

Reply:
Ah I see(not aware of the use case "load un-parsable strings also as NaN" before). It's related with my issue but never mind.